src/fastga_he/models/rta_variables/rta_propulsion_weight.py

```diff
@@ -0,0 +1,146 @@
+            "data:propulsion:he_power_train:CG:x",
+        )
+        self.declare_partials(
+            "data:weight:propulsion:propeller:mass", "data:propulsion:he_power_train:mass", val=1.0
+        )
+
+        if any(
+            component in self.configurator._components_type
+            for component in ["battery_pack", "PEMFC_stack"]
+        ):
+            self.declare_partials(
+                "data:weight:propulsion:engine:mass",
+                "data:propulsion:he_power_train:base_model_engine_mass",
+                val=1.0,
+            )
+            self.declare_partials(
+                "data:weight:propulsion:propeller:mass",
+                "data:propulsion:he_power_train:base_model_engine_mass",
+                val=-1.0,
+            )
+        else:
+            for component_type, component_name in zip(
+                self.propulsive_load_types, self.propulsive_load_names
+            ):
+                if component_type != "PMSM" or "SM_PMSM":
+                    self.declare_partials(
+                        "data:weight:propulsion:engine:mass",
+                        "data:propulsion:he_power_train:"
+                        + component_type
+                        + ":"
+                        + component_name
+                        + ":mass",
+                        val=1.0,
+                    )
+                    self.declare_partials(
+                        "data:weight:propulsion:propeller:mass",
+                        "data:propulsion:he_power_train:"
+                        + component_type
+                        + ":"
+                        + component_name
+                        + ":mass",
+                        val=-1.0,
+                    )
+
+    def compute(self, inputs, outputs, discrete_inputs=None, discrete_outputs=None):
+        powertrain_cg = inputs["data:propulsion:he_power_train:CG:x"]
+        powertrain_weight = inputs["data:propulsion:he_power_train:mass"]
+        engine_weight = 0.0
+
+        if any(
+            component in self.configurator._components_type
+            for component in ["battery_pack", "PEMFC_stack"]
+        ):
+            engine_weight = inputs["data:propulsion:he_power_train:base_model_engine_mass"]
+
+        else:
+            for component_type, component_name in zip(
+                self.propulsive_load_types, self.propulsive_load_names
+            ):
+                if component_type != "PMSM" or "SM_PMSM":
+                    engine_weight += inputs[
+                        "data:propulsion:he_power_train:"
+                        + component_type
+                        + ":"
+                        + component_name
+                        + ":mass"
+                    ]
+
+        outputs["data:weight:propulsion:engine:mass"] = engine_weight
+        outputs["data:weight:propulsion:propeller:mass"] = powertrain_weight - engine_weight
```

Comment:
How is the code dealing with the fact that you have this output which has the same name as another component in the propulsion mass models ?